cmd/utils.go

```diff
@@ -85,3 +88,45 @@ func PrepareMigrationPrerequisites(sourceProfileString, targetProfileString, sou
 	}
 	return sourceProfile, targetProfile, ioHelper, dbName, nil
 }
+
+// MigrateData creates database and populates data in it.
+func MigrateData(ctx context.Context, targetProfile profiles.TargetProfile, sourceProfile profiles.SourceProfile, dbName string, ioHelper *utils.IOStreams, writeLimit int64, conv *internal.Conv, skipForeignKeys, schemaOnly, dataOnly bool) (*writer.BatchWriter, error) {
+	var bw *writer.BatchWriter
+	adminClient, client, dbURI, err := CreateDatabaseClient(ctx, targetProfile, sourceProfile.Driver, dbName, *ioHelper)
+	if err != nil {
+		err = fmt.Errorf("can't create database client: %v", err)
+		return nil, err
+	}
+	defer adminClient.Close()
+	defer client.Close()
+	if !sourceProfile.UseTargetSchema() && dataOnly {
+		err = validateExistingDb(ctx, conv.TargetDb, dbURI, adminClient, client, conv)
+		if err != nil {
+			err = fmt.Errorf("error while validating existing database: %v", err)
+			return nil, err
+		}
+	}
+	if !dataOnly {
+		err = conversion.CreateOrUpdateDatabase(ctx, adminClient, dbURI, sourceProfile.Driver, targetProfile.TargetDb, conv, ioHelper.Out)
+		if err != nil {
+			err = fmt.Errorf("can't create/update database: %v", err)
+			return nil, err
+		}
+	}
+
+	if !schemaOnly {
+		bw, err = conversion.DataConv(ctx, sourceProfile, targetProfile, ioHelper, client, conv, true, writeLimit)
+		if err != nil {
+			err = fmt.Errorf("can't finish data conversion for db %s: %v", dbURI, err)
+			return nil, err
+		}
+
+		if !skipForeignKeys {
+			if err = conversion.UpdateDDLForeignKeys(ctx, adminClient, dbURI, conv, ioHelper.Out); err != nil {
+				err = fmt.Errorf("can't perform update schema on db %s with foreign keys: %v", dbURI, err)
+				return bw, err
+			}
+		}
```

Comment:
This seems like too much branching, it will be hard to unit test and cover the branches for this method, especially if we add more subcommands. Can you think of a way to solve this?

Reply:
updated